infra/src/test/kotlin/fi/fta/geoviite/infra/geometry/VerticalGeometryListingFileDaoIT.kt

```diff
@@ -0,0 +1,57 @@
+package fi.fta.geoviite.infra.geometry
+
+import fi.fta.geoviite.infra.DBTestBase
+import fi.fta.geoviite.infra.util.FileName
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import java.time.Instant
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+@ActiveProfiles("dev", "test")
+@SpringBootTest
+class VerticalGeometryListingFileDaoIT @Autowired constructor(
+    private val verticalGeometryListingFileDao: VerticalGeometryListingFileDao,
+): DBTestBase() {
+
+    @BeforeEach
+    fun setUp() {
+        jdbc.update("delete from layout.vertical_geometry_listing_file where id = 1", mapOf<String, Any>())
+    }
+
+    @Test
+    fun `Entire track network vertical geometry file is correctly upserted and fetched`() {
+        assertNull(verticalGeometryListingFileDao.getVerticalGeometryListingFile())
+        assertEquals(Instant.EPOCH, verticalGeometryListingFileDao.getLastFileListingTime())
+
+        val originalFile = VerticalGeometryListingFile(
+            name = FileName("vertical geometry test file name 1"),
+            content = """
+                vert_geom_header1, vert_geom_header2, vert_geom_header3
```

Comment:
Nämä columnien nimet ja arvot näyttää vähän jänniltä näissä. Eihän tänne jäänyt mitään placeholderia?

Reply:
Kyseessä siis vastaavantyyppinen testi kuin täällä: https://github.com/finnishtransportagency/geoviite/blob/main/infra/src/test/kotlin/fi/fta/geoviite/infra/tracklayout/ElementListingFileDaoIT.kt#L23

Suosin itse testeissä ns. testikohtaisesti uniikkien stringien yms käyttöä. Näin testibugien paikantaminen menee helpommaksi.

 Tässä testiversiossa varsinaisella sisällöllä ei pitäisi olla merkitystä, sillä tätähän ei esimerkiksi validoida CSV-tiedostoksi.